datafusion/physical-plan/src/sorts/stream.rs

```diff
@@ -127,7 +127,7 @@ impl RowCursorStream {
         // track the memory in the newly created Rows.
         let mut rows_reservation = self.reservation.new_empty();
         rows_reservation.try_grow(rows.size())?;
-        Ok(RowValues::new(rows, rows_reservation))
+        Ok(RowValues::new(Arc::new(rows)))
```

Comment:
The memory reservation is still grown on the line above.
Since the `RowValues` take Arc::rows, it's not tied to the memory reservation. (Meaning, on the RowValue slicing is not adding to the reservation.)